src/Akka.Persistence.Sql.Linq2Db/Journal/DAO/BaseByteArrayJournalDao.cs

```diff
@@ -390,32 +392,34 @@ public override
                 }
 
                 
-                //TODO: Is there a better way to do this async?
-                //return Source
-                //    .FromObservable(query.AsAsyncEnumerable().ToObservable())
-                
-                return Source.From(await query.ToListAsync())
+                return AsyncSource<JournalRow>.FromEnumerable(query,async q=>await q.ToListAsync())
                     .Via(
-                        Serializer.DeserializeFlow()).Select(sertry =>
-                    {
-                        if (sertry.IsSuccess)
-                        {
-                            return new
-                                Util.Try<ReplayCompletion>(
-                                    new ReplayCompletion()
-                                    {
-                                        repr = sertry.Success.Value.Item1,
-                                        Ordering = sertry.Success.Value.Item3
-                                    });
-                        }
-                        else
-                        {
-                            return new
-                                Util.Try<ReplayCompletion>(
-                                    sertry.Failure.Value);
-                        }
-                    });
+                        deserializeFlow).Select(MessageWithBatchMapper());
             }
         }
+
+        private static Func<Util.Try<(IPersistentRepresentation, IImmutableSet<string>, long)>, Util.Try<ReplayCompletion>> MessageWithBatchMapper()
```

Comment:
Somehow, this was chosen by refactoring tools as the 'best' way to refactor with minimal allocations. I'm not sure that's true.